### ISSUE TITLE
Erroring-out camera sources cause working sources not to be found

### ISSUE THREAD

**Issue by RastislavKish** [flying-pizza-69/GestureX#7]
First of all, also a Reddit visitor, thanks for the super-cool project! :)

I'm not sure how to write reproduction steps for this, but basically, I have my laptop internal webcam covered by duct tape, and use an external-one when necessary. However, my system also has some special video sources hanging around, probably streaming loopbacks I set up in the past for OBS, and OpenCV can't open and read these. That on its own wouldn't be an issue, but what GestureX is doing right now is using the hit and miss strategy for finding working cameras, trying out increasing id numbers, and the respective loop breaks when OpenCV is unable to read the id. However, on my system, camera 0 works fine, camera 1 errors out and camera 2 also works fine. But GestureX can only find camera 0 due to the algorithm.

Everything works when I remove the break if statement branch. Perhaps this should be the default behavior of the algorithm, users may have permission-restricted or malfunctioning cameras on their system, which would be preventing the available-ones from being found, like in my case.

One downside right now is that OpenCV seems to print out an error log into the terminal when there's problem opening a camera, with the current way of handling things there's just one, if all ids from 0 to 10 were tried, there would be 9 errors with most configurations, which is not a particularly fancy output, although it should be rather a minor nuissance.

**Comment by flying-pizza-69** [comment]
you are right, removing the break system fixes the issue of not being able to switch camera sources. let me know if it works for you now.

yeah the current way of checking available camera sources is kinda awful haha. i can look for suppressing cv2 warnings but im planning to find a better solution than that, maybe using some dedicated libraries like pyv4l2. i will keep this issue opened till i fix the warn/error issue.

thanks!

**Comment by RastislavKish** [comment]
Thanks! Yes, now it works properly for me. Here is the log when running b54894c:

```
2024-04-18 00:09:52.155187: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-04-18 00:09:52.155248: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-04-18 00:09:52.156464: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-04-18 00:09:52.163624: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-04-18 00:09:53.115602: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
/home/username/Development/Git/GestureX/gui.py:101: DeprecationWarning: Gtk.Widget.override_color is deprecated
  separator.override_color(Gtk.StateFlags.NORMAL, Gdk.RGBA(0, 0, 0, 1))
[ WARN:0@4,180] global cap_v4l.cpp:997 open VIDEOIO(V4L2:/dev/video1): can't open camera by index
[ERROR:0@4,296] global obsensor_uvc_stream_channel.cpp:159 getStreamChannelGroup Camera index out of range
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1713391795.885621  119592 gl_context_egl.cc:85] Successfully initialized EGL. Major : 1 Minor: 5
I0000 00:00:1713391795.888307  119599 gl_context.cc:357] GL version: 3.2 (OpenGL ES 3.2 Mesa 23.2.1-1ubuntu3.1~22.04.2), renderer: Mesa Intel(R) HD Graphics 4400 (HSW GT2)
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
```

Interestingly, in /dev, I seem to have four video devices (0-3), only 0 and 2 are picked up by OpenCV. Honestly, I have no idea what the other two are, it was some time since I messed with this stuff. :)
